net/icmpv6/icmpv6_reply.c

```diff
@@ -77,14 +77,13 @@
  *   dev   - The device driver structure containing the received packet
  *   type  - ICMPv6 Message Type, eg. ICMPv6_DEST_UNREACHABLE
  *   code  - ICMPv6 Message Code, eg. ICMPv6_PORT_UNREACH
- *   data  - Additional 32-bit parameter in the ICMPv6 header
  *
  * Returned Value:
  *   None
  *
  ****************************************************************************/
 
-void icmpv6_reply(FAR struct net_driver_s *dev, int type, int code, int data)
```

Comment:
I think it is better to keep the data argument, which will be used in other message types in the future

3.2.  Packet Too Big Message
3.4.  Parameter Problem Message

https://datatracker.ietf.org/doc/html/rfc4443

Reply:
ack, I see two builds also failed because I changed the api. What's the best way to unpack the value? I could treat it like the `len` field for now, but I'm concerned that it's wrong wrong byte order (and it's impossible to test with zero values).

Note that https://datatracker.ietf.org/doc/html/rfc4884 adds data args to Destination Unreachable (and others too).

Reply:
Updated to remove the argument change and use the same logic that len, ipid and ipoffset use to split the `uint32_t` into the two `uint16_t` fields.